src/lib/executeSourceTransfer.js

```diff
@@ -1,88 +0,0 @@
-'use strict'
-
-const log = require('../common').log.create('executeSourceTransfer')
-const validator = require('./validate')
-const moment = require('moment')
-const _ = require('lodash')
-const promiseRetry = require('promise-retry')
-const retryOpts = {
-  forever: true,
-  factor: 2,
-  minTimeout: 10
-}
-
-// Add the execution_condition_fulfillment to the source transfer
-// and submit it to the source ledger
-async function executeSourceTransfer (destinationTransfer, fulfillment, fulfillmentData, ledgers, backend, config) {
-  if (!fulfillment) {
-    log.error('Cannot execute source transfers, no condition fulfillment found. Destination transfer: ' + JSON.stringify(destinationTransfer))
-    return
-  }
-
-  const noteToSelf = destinationTransfer.noteToSelf || {}
-  const sourceTransferLedger = noteToSelf.source_transfer_ledger
-  const sourceTransferID = noteToSelf.source_transfer_id
-  const sourceTransferAmount = noteToSelf.source_transfer_amount
-  validator.validate('IlpAddress', sourceTransferLedger)
-  validator.validate('Uuid', sourceTransferID)
-  validator.validate('Amount', sourceTransferAmount)
-
-  log.debug('Requesting fulfillment of source transfer: ' + sourceTransferID + ' (fulfillment: ' + JSON.stringify(fulfillment) + ')')
-  const plugin = ledgers.getPlugin(sourceTransferLedger)
-
-  const sourceTransferExpiry = moment(destinationTransfer.expiresAt, moment.ISO_8601)
-    .add(config.expiry.minMessageWindow, 'seconds')
-  const retryTimeout = sourceTransferExpiry.diff(moment())
-
-  const timeoutPromise = new Promise((resolve, reject) => {
-    setTimeout(function () {
-      reject(`Fulfillment of source transfer ${sourceTransferID} timed out.`)
-    }, retryTimeout)
-  })
-
-  const fulfillPromise = promiseRetry(retryOpts, function (retry, number) {
-    return Promise.resolve(plugin.fulfillCondition(sourceTransferID, fulfillment, fulfillmentData))
-      .catch(function (err) {
-        if (shouldRetry(err)) {
```

Comment:
Should we add fulfillment retry logic to the `ilp-compat-plugin`?

Reply:
I do think that fulfillment retries should be in the plugin because it depends on the ledger protocol if and how you want to retry.

Should it be in ilp-plugin-compat? I don't think it's worth the effort. Retry logic is an optimization, it's only to minimize unnecessary connector losses. I'd say we migrate to LPI2 and then add retry logic per plugin. Different plugins may want to handle retries quite differently, depending on whether they're TCP or UDP based, session or request based or to deal with some other ledger/transport-specific quirks.